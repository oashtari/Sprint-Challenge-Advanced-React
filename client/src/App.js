import React from 'react';
import logo from './logo.svg';
import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import axios from 'axios';

import PlayerCard from './Components/WWCplayersCard';

class App extends React.Component {


  constructor() {
    super();
    this.state = {
      players: []
      // name: '',
      // country: '',
      // searches: '',
      // id: ''

    }
  }


  componentDidMount() {
    axios
      .get(`http://localhost:5000/api/players`)
      .then(res => {

        this.setState({ ...this.state, players: res.data })

      })
      .catch(err => console.log(err))
  }




  render() {
    return (
      <div className="App" >
        <div className='wwc_players'>
          {/* <FilterPlayers props={this.state.playersList} /> */}
          <h1> WWC Players:</h1 >
          {this.state.players.map(player => (
            < PlayerCard player={player} />
          ))}
        </div>
      </div>
    );
  }
}


export default App;
