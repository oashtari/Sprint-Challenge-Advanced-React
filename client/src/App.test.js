import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import WWCPlayers from './Components/WWCplayers';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('renders without crashing', () => {
  render(<App />);
})

test('contains WWC Players in display', () => {

  const container = render(<App />);
  const getByText = container.getByText;

  getByText(/wwc players/i);
});


test('matches the snapshot', () => {
  const tree = renderer.create(<WWCPlayers />).toJSON()

  expect(tree).toMatchSnapshot()
});


// test("all players", () => {
//   const { getByTestId } = render(<WWCPlayers />);

//   getByTestId("allPlayers");
// });