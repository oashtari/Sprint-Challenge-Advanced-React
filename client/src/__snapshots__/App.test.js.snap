// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="players"
>
  <button
    aria-label={null}
    className="btn btn-danger"
    onClick={[Function]}
    type="button"
  >
    Store Players
  </button>
</div>
`;
